shell/renderer/api/electron_api_web_frame.cc

```diff
@@ -130,10 +134,54 @@ class ScriptExecutionCallback : public blink::WebScriptExecutionCallback {
     v8::Isolate* isolate = v8::Isolate::GetCurrent();
     if (!result.empty()) {
       if (!result[0].IsEmpty()) {
-        // Right now only single results per frame is supported.
-        if (!callback_.is_null())
-          std::move(callback_).Run(result[0], v8::Undefined(isolate));
-        promise_.Resolve(result[0]);
+        // Either world safe results are disabled or the result was created in
+        // the same world as the caller
+        if (!world_safe_result_ ||
+            (result[0]->IsObject() &&
+             promise_.GetContext() ==
+                 result[0].As<v8::Object>()->CreationContext())) {
```

Comment:
if `result[0]` is a `Primitive` can we return it directly?

Reply:
No. Primitives still have prototypes right?

Reply:
I don't think so? `GetPrototype` is defined on `v8::Object`.